os/kernel/binary_manager/binary_manager_bootparam.c

```diff
@@ -0,0 +1,147 @@
+/****************************************************************************
+ *
+ * Copyright 2021 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+#include <tinyara/config.h>
+#include <debug.h>
+#include <stdio.h>
+#include <string.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <stdbool.h>
+
+#include <tinyara/binary_manager.h>
+
+#include "binary_manager.h"
+
+/* Data for Boot parameters */
+static binmgr_bpinfo_t bp_info;
```

Comment:
This static variable seems to be accessed by several APIs each of which can be called by separate thread.
How can this structure be protected from accessing by multiple threads?

Reply:
Before refactoring about separation of BM operations like loading, reloading to each threads, binary manager only accessed this variable so any protection was unnecessary.
But now this can be accessed as you said and should be protected.
I will add codes for protection with another PR. Thank you for good point I missed.

Reply:
Do you guarantee that only the bin manager (i.e., a single thread) manipulate this variable?

Reply:
Yes. I guarantee that with current implementation.
There are many things to be implemented about bootparam.
I will check and add codes for protection if it is needed at that time.